# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/EventMessageList.h
# Commit: 5718f7f3067e
# Full Hash: 5718f7f3067eff42e097a6013acd82fbe6aa299a
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2025-05-15 22:58:00
# Regressor Bug: 1550462
# File Overlap Count: 1
# Description:
#   Bug 1550462 - part 2: Make `PresShell::HandleEvent` dispatch preceding `pointerrawupdate` event r=smaug,dom-core,edgar
#   
#   This patch tries to dispatch `ePointerRawUpdate` with
#   `PresShell::EventHandler::DispatchPrecedingPointerEvent` as same as usual
#   pointer events.
# ==============================================================================

diff -r aae164cff6aa -r 5718f7f3067e widget/EventMessageList.h
--- a/widget/EventMessageList.h	Thu May 15 03:20:28 2025 +0000
+++ b/widget/EventMessageList.h	Thu May 15 03:20:29 2025 +0000
@@ -93,6 +93,14 @@
 NS_EVENT_MESSAGE(eMouseLeave)
 NS_EVENT_MESSAGE(eMouseTouchDrag)
 NS_EVENT_MESSAGE(eMouseLongTap)
+// eMouseRawUpdate is for dispatching ePointerRawUpdate caused by a mouse input.
+// When we dispatch ePointerRawUpdate, we need to handle the steps defined by
+// the spec in "fire a pointer event" section like the other pointer event.
+// The steps are handled by
+// PresShell::EventHandler::DispatchPrecedingPointerEvent() and for using it, we
+// should dispatch this internal event instead of dispatching ePointerRawUpdate
+// directly.
+NS_EVENT_MESSAGE(eMouseRawUpdate)
 NS_EVENT_MESSAGE(eMouseExploreByTouch)
 NS_EVENT_MESSAGE_FIRST_LAST(eMouseEvent, eMouseMove, eMouseExploreByTouch)
 
@@ -100,6 +108,13 @@
 NS_EVENT_MESSAGE(ePointerAuxClick)
 
 // Pointer spec events
+// NOTE: We handle the steps before dispatching a pointer event which is defined
+// by the spec in "fire a pointer event" section in
+// PresShell::EventHandler::DispatchPrecedingPointerEvent().  Therefore,
+// we should not dispatch ePointer* events with
+// `PresShell::EventHandler::HandleEvent` directly.  Create a new internal
+// message for implementing a new pointer event if the new event is defined as
+// dispatched with "fire a pointer event" steps.
 NS_EVENT_MESSAGE(ePointerMove)
 NS_EVENT_MESSAGE(ePointerUp)
 NS_EVENT_MESSAGE(ePointerDown)
@@ -433,6 +448,14 @@
 NS_EVENT_MESSAGE(eTouchEnd)
 NS_EVENT_MESSAGE(eTouchCancel)
 NS_EVENT_MESSAGE(eTouchPointerCancel)
+// eTouchRawUpdate is for dispatching ePointerRawUpdate caused by a touch.
+// When we dispatch ePointerRawUpdate, we need to handle the steps defined by
+// the spec in "fire a pointer event" section like the other pointer event.
+// The steps are handled by
+// PresShell::EventHandler::DispatchPrecedingPointerEvent() and for using it, we
+// should dispatch this internal event instead of dispatching ePointerRawUpdate
+// directly.
+NS_EVENT_MESSAGE(eTouchRawUpdate)
 
 // Pointerlock DOM API
 NS_EVENT_MESSAGE(ePointerLockChange)